.github/workflows/test.yml

```diff
@@ -3,11 +3,21 @@ name: test
 on: [push, pull_request]
 
 jobs:
+  ruby-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      versions: ${{ steps.versions.outputs.value }}
+    steps:
+      - id: versions
+        run: |
+          versions=$(curl -s 'https://cache.ruby-lang.org/pub/misc/ci_versions.json' | jq -c)
+          echo "::set-output name=value::${versions}"
   build:
+    needs: ruby-versions
     name: build (${{ matrix.ruby }} / ${{ matrix.os }})
     strategy:
       matrix:
-        ruby: [ '3.0', 2.7, 2.6, 2.5, 2.4, head ]
+        ruby: ${{fromJson(needs.ruby-versions.outputs.versions)}}
```

Comment:
Perhaps it might be better to add spaces in `${{ .. }}`, such as `${{ fromJson(needs.ruby-versions.outputs.versions) }}`.

Reply:
👍  Thanks, applied.